File: Java/src/main/java/lib/Tabla.java

package lib;

/**
 *
 * 
 */
public class Tabla {

    public void generarTabla(String[] encabezados, String[][] datos) {
        System.out.printf("\n\t");
	for (int i = 0; i < encabezados.length; i++) {
		System.out.printf("|-----------------------------------------------------------");	
	}
	System.out.printf("%s", "|");

	System.out.printf("\n\t|");
	for (int i = 0; i < encabezados.length; i++) {
		System.out.printf("%30s%30s",encabezados[i], "|");	
	}

	System.out.printf("\n\t|");
	for (int i = 0; i < encabezados.length; i++) {
		System.out.printf("-----------------------------------------------------------|");	
	}

	// Imprimiendo la lista de datos
	for (int i = 0; i < datos.length; i++) {
		
		System.out.printf("\n\t|");
		for (int j = 0; j < datos[0].length; j++) {
			System.out.printf("%30s%30s",datos[i][j], "|");	
		}

		System.out.printf("\n\t|");
		for (int j = 0; j < datos[0].length; j++) {
			System.out.printf("-----------------------------------------------------------|");			
		}
	}
	System.out.printf("\n");
    }
}

File: Go/tabla/tabla.go
package tabla

import "fmt"

func GenerarTabla(encabezados []string, datos [][]string) {

	fmt.Printf("\n\t")
	for i := 0; i < len(encabezados); i++ {
		fmt.Printf("|-----------------------------------------------------------")	
	}
	fmt.Printf("%s", "|")

	fmt.Printf("\n\t|")
	for i := 0; i < len(encabezados); i++ {
		fmt.Printf("%30s%30s",encabezados[i], "|")	
	}

	fmt.Printf("\n\t|")
	for i := 0; i < len(encabezados); i++ {
		fmt.Printf("-----------------------------------------------------------|")	
	}

	// Imprimiendo la lista de datos
	for i := 0; i < len(datos); i++ {
		
		fmt.Printf("\n\t|")
		for j := 0; j < len(datos[0]); j++ {
			fmt.Printf("%30s%30s",datos[i][j], "|")	
		}

		fmt.Printf("\n\t|")
		for j := 0; j < len(datos[0]); j++ {
			fmt.Printf("-----------------------------------------------------------|")			
		}
	}
	fmt.Printf("\n")
}